firefox-ios/Client/Frontend/Browser/TemporaryDocument.swift

```diff
@@ -36,6 +41,41 @@ class TemporaryDocument: NSObject {
         }
     }
 
+    /// Uses modern concurrency to download the file and save to temporary storage.
```

Comment:
@lmarceau FYI I had to fix some xcode 16.2 /Swift 6 issues in Bitrise from `@Sendable`, so I just rolled that into my FXIOS-10830 fix which I needed to do next anyway.